amundsen_application/static/js/components/DashboardPage/ImagePreview/index.spec.tsx

```diff
@@ -71,29 +72,80 @@ describe('ImagePreview', () => {
         });
       });
 
-      describe('when not loading', () => {
+      describe('when loaded', () => {
         let props;
         let wrapper;
+
         beforeAll(() => {
           const setupResult = setup();
           props = setupResult.props
           wrapper = setupResult.wrapper;
-          wrapper.instance().setState({ isLoading: false, hasError:false });
+          wrapper.instance().setState({ isLoading: false, hasError: false });
         });
+
         it('renders visible img with correct props', () => {
           const elementProps = wrapper.find('img').props();
-          expect(elementProps.style).toEqual({ visibility: 'visible' });
+
           expect(elementProps.src).toEqual(`${Constants.PREVIEW_BASE}/${props.uri}/${Constants.PREVIEW_END}`);
           expect(elementProps.onLoad).toBe(wrapper.instance().onSuccess);
           expect(elementProps.onError).toBe(wrapper.instance().onError);
         });
+
+        it('renders a button', () => {
+          const expected = 1;
+          const actual = wrapper.find('.preview-button').length;
+
+          expect(actual).toEqual(expected);
+        });
       })
     });
 
-    it('renders link if hasError', () => {
-      const { props, wrapper } = setup();
-      wrapper.instance().setState({ hasError: true });
-      expect(wrapper.find(Linkify).exists()).toBeTruthy();
+    describe('when there is an error', () => {
+      it('renders a link', () => {
+        const { props, wrapper } = setup();
+
+        wrapper.instance().setState({ hasError: true });
+        wrapper.update();
```

Comment:
Do we need this update, or is it a best practice? Does `setState` alone risk not functioning as expected?

Reply:
Yes, is a best practice to ensure you get a render before running the expectation. As setState is asynchronous, it could happen. Probably not in this case, but just in case.

Reply:
This was also breaking when using "mount", as it wasn't finding the Linkify component on the rendered tree.

Somehow, shallow gives a false positive there, like it was rendered, but it was only "supposed to render".

Reply:
Didn't expect that. Good to know.